lib/mongoid/matchable/default.rb

```diff
@@ -69,6 +69,32 @@ def determine(value, operator)
           attr ? attr.send(operator, first(value)) : false
         }
       end
+
+      # Convenience method for checking _matches? on a Document or a Hash.
+      #
+      # @example Does the document of type Document or Hash match?
+      #   safe_matches?(default, "a", 1)
+      #   safe_matches?(:a => 1, "b", 2)
+      #
+      # @param [ Document, Hash ] document The object of type Document or Hash to call _matches? on
+      # @param [ String ] key The key
+      # @param [ Object ] value The value to check if it matches.
+      #
+      # @return [ true, false ] True if matches, false if not.
+      #
+      # @since 7.0.4
+      def safe_matches?(document, key, value)
+        if document.is_a?(Document)
+          document._matches?(key => value)
+        else
+          # If not a Document, then assume it's a Hash
+          if value.try(:first).try(:[],0) == "$not".freeze || value.try(:first).try(:[],0) == :$not
```

Comment:
what if `$not` isn't the first key in the hash? Don't you need to iterate through all the keys?